### ISSUE TITLE
Incompatible with Magento 2.4.6 | Zend framework error

### ISSUE THREAD

**Issue by gerrits-ecommerce** [dpdconnect/magento2-shipping#71]
After updating our Magento 2 instance to the latest version 2.4.6 the following error occurs when creating a shipment in the Magento 2 backend.

![image](https://user-images.githubusercontent.com/31535836/235632339-e526bfc1-cc20-46a8-9d85-893303166ec2.png)

```
<option value="Error: Class "Zend_Measure_Weight" not found in /var/www/vendor/dpdconnect/magento2-shipping/view/adminhtml/templates/order/packaging/popup_content.phtml:154
Stack trace:
#0 /var/www/vendor/dpdconnect/magento2-shipping/view/adminhtml/templates/order/packaging/popup.phtml(391): include()
#1 /var/www/vendor/magento/framework/View/TemplateEngine/Php.php(71): include('/var/www/vendor...')
#2 /var/www/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\TemplateEngine\Php->render(Object(DpdConnect\Shipping\Block\Adminhtml\Order\Packaging\Interceptor), '/var/www/vendor...', Array)
#3 /var/www/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\TemplateEngine\Php\Interceptor->___callParent('render', Array)
#4 /var/www/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\TemplateEngine\Php\Interceptor->Magento\Framework\Interception\{closure}(Object(DpdConnect\Shipping\Block\Adminhtml\Order\Packaging\Interceptor), '/var/www/vendor...', Array)
#5 /var/www/generated/code/Magento/Framework/View/TemplateEngine/Php/Interceptor.php(23): Magento\Framework\View\TemplateEngine\Php\Interceptor->___callPlugins('render', Array, Array)
#6 /var/www/vendor/magento/framework/View/Element/Template.php(263): Magento\Framework\View\TemplateEngine\Php\Interceptor->render(Object(DpdConnect\Shipping\Block\Adminhtml\Order\Packaging\Interceptor), '/var/www/vendor...', Array)
#7 /var/www/generated/code/DpdConnect/Shipping/Block/Adminhtml/Order/Packaging/Interceptor.php(329): Magento\Framework\View\Element\Template->fetchView('/var/www/vendor...')
#8 /var/www/vendor/magento/framework/View/Element/Template.php(293): DpdConnect\Shipping\Block\Adminhtml\Order\Packaging\Interceptor->fetchView('/var/www/vendor...')
#9 /var/www/vendor/magento/module-backend/Block/Template.php(141): Magento\Framework\View\Element\Template->_toHtml()
#10 /var/www/vendor/magento/framework/View/Element/AbstractBlock.php(1095): Magento\Backend\Block\Template->_toHtml()
#11 /var/www/vendor/magento/framework/View/Element/AbstractBlock.php(1099): Magento\Framework\View\Element\AbstractBlock->Magento\Framework\View\Element\{closure}()
#12 /var/www/vendor/magento/framework/View/Element/AbstractBlock.php(660): Magento\Framework\View\Element\AbstractBlock->_loadCache()
#13 /var/www/generated/code/DpdConnect/Shipping/Block/Adminhtml/Order/Packaging/Interceptor.php(554): Magento\Framework\View\Element\AbstractBlock->toHtml()
#14 /var/www/vendor/magento/framework/View/Layout.php(578): DpdConnect\Shipping\Block\Adminhtml\Order\Packaging\Interceptor->toHtml()
#15 /var/www/vendor/magento/framework/View/Layout.php(555): Magento\Framework\View\Layout->_renderBlock('shipment_packag...')
#16 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('shipment_packag...')
#17 /var/www/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('shipment_packag...')
#18 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('shipment_packag...', true)
#19 /var/www/vendor/magento/framework/View/Element/AbstractBlock.php(507): Magento\Framework\View\Layout\Interceptor->renderElement('shipment_packag...', true)
#20 /var/www/generated/code/Magento/Shipping/Block/Adminhtml/View/Form/Interceptor.php(437): Magento\Framework\View\Element\AbstractBlock->getChildHtml('shipment_packag...', true)
#21 /var/www/vendor/magento/module-shipping/view/adminhtml/templates/view/form.phtml(89): Magento\Shipping\Block\Adminhtml\View\Form\Interceptor->getChildHtml('shipment_packag...')
#22 /var/www/vendor/magento/framework/View/TemplateEngine/Php.php(71): include('/var/www/vendor...')
#23 /var/www/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\Shipping\Block\Adminhtml\View\Form\Interceptor), '/var/www/vendor...', Array)
#24 /var/www/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\TemplateEngine\Php\Interceptor->___callParent('render', Array)
#25 /var/www/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\TemplateEngine\Php\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Shipping\Block\Adminhtml\View\Form\Interceptor), '/var/www/vendor...', Array)
#26 /var/www/generated/code/Magento/Framework/View/TemplateEngine/Php/Interceptor.php(23): Magento\Framework\View\TemplateEngine\Php\Interceptor->___callPlugins('render', Array, Array)
#27 /var/www/vendor/magento/framework/View/Element/Template.php(263): Magento\Framework\View\TemplateEngine\Php\Interceptor->render(Object(Magento\Shipping\Block\Adminhtml\View\Form\Interceptor), '/var/www/vendor...', Array)
#28 /var/www/generated/code/Magento/Shipping/Block/Adminhtml/View/Form/Interceptor.php(275): Magento\Framework\View\Element\Template->fetchView('/var/www/vendor...')
#29 /var/www/vendor/magento/framework/View/Element/Template.php(293): Magento\Shipping\Block\Adminhtml\View\Form\Interceptor->fetchView('/var/www/vendor...')
#30 /var/www/vendor/magento/module-backend/Block/Template.php(141): Magento\Framework\View\Element\Template->_toHtml()
#31 /var/www/vendor/magento/framework/View/Element/AbstractBlock.php(1095): Magento\Backend\Block\Template->_toHtml()
#32 /var/www/vendor/magento/framework/View/Element/AbstractBlock.php(1099): Magento\Framework\View\Element\AbstractBlock->Magento\Framework\View\Element\{closure}()
#33 /var/www/vendor/magento/framework/View/Element/AbstractBlock.php(660): Magento\Framework\View\Element\AbstractBlock->_loadCache()
#34 /var/www/generated/code/Magento/Shipping/Block/Adminhtml/View/Form/Interceptor.php(500): Magento\Framework\View\Element\AbstractBlock->toHtml()
#35 /var/www/vendor/magento/framework/View/Layout.php(578): Magento\Shipping\Block\Adminhtml\View\Form\Interceptor->toHtml()
#36 /var/www/vendor/magento/framework/View/Layout.php(555): Magento\Framework\View\Layout->_renderBlock('form')
#37 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('form')
#38 /var/www/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('form')
#39 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('form', true)
#40 /var/www/vendor/magento/framework/View/Element/AbstractBlock.php(507): Magento\Framework\View\Layout\Interceptor->renderElement('form', true)
#41 /var/www/generated/code/Magento/Shipping/Block/Adminhtml/View/Interceptor.php(473): Magento\Framework\View\Element\AbstractBlock->getChildHtml('form', true)
#42 /var/www/vendor/magento/module-backend/Block/Widget/Form/Container.php(220): Magento\Shipping\Block\Adminhtml\View\Interceptor->getChildHtml('form')
#43 /var/www/generated/code/Magento/Shipping/Block/Adminhtml/View/Interceptor.php(104): Magento\Backend\Block\Widget\Form\Container->getFormHtml()
#44 /var/www/vendor/magento/module-backend/view/adminhtml/templates/widget/form/container.phtml(25): Magento\Shipping\Block\Adminhtml\View\Interceptor->getFormHtml()
#45 /var/www/vendor/magento/framework/View/TemplateEngine/Php.php(71): include('/var/www/vendor...')
#46 /var/www/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\View\TemplateEngine\Php->render(Object(Magento\Shipping\Block\Adminhtml\View\Interceptor), '/var/www/vendor...', Array)
#47 /var/www/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\View\TemplateEngine\Php\Interceptor->___callParent('render', Array)
#48 /var/www/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\View\TemplateEngine\Php\Interceptor->Magento\Framework\Interception\{closure}(Object(Magento\Shipping\Block\Adminhtml\View\Interceptor), '/var/www/vendor...', Array)
#49 /var/www/generated/code/Magento/Framework/View/TemplateEngine/Php/Interceptor.php(23): Magento\Framework\View\TemplateEngine\Php\Interceptor->___callPlugins('render', Array, Array)
#50 /var/www/vendor/magento/framework/View/Element/Template.php(263): Magento\Framework\View\TemplateEngine\Php\Interceptor->render(Object(Magento\Shipping\Block\Adminhtml\View\Interceptor), '/var/www/vendor...', Array)
#51 /var/www/generated/code/Magento/Shipping/Block/Adminhtml/View/Interceptor.php(311): Magento\Framework\View\Element\Template->fetchView('/var/www/vendor...')
#52 /var/www/vendor/magento/framework/View/Element/Template.php(293): Magento\Shipping\Block\Adminhtml\View\Interceptor->fetchView('/var/www/vendor...')
#53 /var/www/vendor/magento/module-backend/Block/Template.php(141): Magento\Framework\View\Element\Template->_toHtml()
#54 /var/www/vendor/magento/framework/View/Element/AbstractBlock.php(1095): Magento\Backend\Block\Template->_toHtml()
#55 /var/www/vendor/magento/framework/View/Element/AbstractBlock.php(1099): Magento\Framework\View\Element\AbstractBlock->Magento\Framework\View\Element\{closure}()
#56 /var/www/vendor/magento/framework/View/Element/AbstractBlock.php(660): Magento\Framework\View\Element\AbstractBlock->_loadCache()
#57 /var/www/generated/code/Magento/Shipping/Block/Adminhtml/View/Interceptor.php(536): Magento\Framework\View\Element\AbstractBlock->toHtml()
#58 /var/www/vendor/magento/framework/View/Layout.php(578): Magento\Shipping\Block\Adminhtml\View\Interceptor->toHtml()
#59 /var/www/vendor/magento/framework/View/Layout.php(555): Magento\Framework\View\Layout->_renderBlock('sales_shipment_...')
#60 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('sales_shipment_...')
#61 /var/www/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('sales_shipment_...')
#62 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('sales_shipment_...', false)
#63 /var/www/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('sales_shipment_...', false)
#64 /var/www/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('content', false)
#65 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('content')
#66 /var/www/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('content')
#67 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('content', false)
#68 /var/www/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('content', false)
#69 /var/www/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('main.col', false)
#70 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('main.col')
#71 /var/www/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('main.col')
#72 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('main.col', false)
#73 /var/www/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('main.col', false)
#74 /var/www/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('admin.scope.col...', false)
#75 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('admin.scope.col...')
#76 /var/www/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('admin.scope.col...')
#77 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('admin.scope.col...', false)
#78 /var/www/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('admin.scope.col...', false)
#79 /var/www/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('page.main.conta...', false)
#80 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('page.main.conta...')
#81 /var/www/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('page.main.conta...')
#82 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('page.main.conta...', false)
#83 /var/www/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('page.main.conta...', false)
#84 /var/www/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('page.content', false)
#85 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('page.content')
#86 /var/www/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('page.content')
#87 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('page.content', false)
#88 /var/www/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('page.content', false)
#89 /var/www/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('page.wrapper', false)
#90 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('page.wrapper')
#91 /var/www/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('page.wrapper')
#92 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('page.wrapper', false)
#93 /var/www/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('page.wrapper', false)
#94 /var/www/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('backend.page', false)
#95 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('backend.page')
#96 /var/www/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('backend.page')
#97 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('backend.page', false)
#98 /var/www/vendor/magento/framework/View/Layout.php(606): Magento\Framework\View\Layout\Interceptor->renderElement('backend.page', false)
#99 /var/www/vendor/magento/framework/View/Layout.php(557): Magento\Framework\View\Layout->_renderContainer('root', false)
#100 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(149): Magento\Framework\View\Layout->renderNonCachedElement('root')
#101 /var/www/vendor/magento/framework/View/Layout.php(510): Magento\Framework\View\Layout\Interceptor->renderNonCachedElement('root')
#102 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(140): Magento\Framework\View\Layout->renderElement('root', true)
#103 /var/www/vendor/magento/framework/View/Layout.php(975): Magento\Framework\View\Layout\Interceptor->renderElement('root')
#104 /var/www/generated/code/Magento/Framework/View/Layout/Interceptor.php(347): Magento\Framework\View\Layout->getOutput()
#105 /var/www/vendor/magento/framework/View/Result/Page.php(260): Magento\Framework\View\Layout\Interceptor->getOutput()
#106 /var/www/vendor/magento/framework/View/Result/Layout.php(171): Magento\Framework\View\Result\Page->render(Object(Magento\Framework\App\Response\Http\Interceptor))
#107 /var/www/generated/code/Magento/Backend/Model/View/Result/Page/Interceptor.php(140): Magento\Framework\View\Result\Layout->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#108 /var/www/vendor/magento/framework/App/Http.php(120): Magento\Backend\Model\View\Result\Page\Interceptor->renderResult(Object(Magento\Framework\App\Response\Http\Interceptor))
#109 /var/www/vendor/magento/framework/Interception/Interceptor.php(58): Magento\Framework\App\Http->launch()
#110 /var/www/vendor/magento/framework/Interception/Interceptor.php(138): Magento\Framework\App\Http\Interceptor->___callParent('launch', Array)
#111 /var/www/vendor/magento/framework/Interception/Interceptor.php(153): Magento\Framework\App\Http\Interceptor->Magento\Framework\Interception\{closure}()
#112 /var/www/generated/code/Magento/Framework/App/Http/Interceptor.php(23): Magento\Framework\App\Http\Interceptor->___callPlugins('launch', Array, Array)
#113 /var/www/vendor/magento/framework/App/Bootstrap.php(264): Magento\Framework\App\Http\Interceptor->launch()
#114 /var/www/pub/index.php(30): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor))
#115 {main}
```

It looks like you the module using a Zend framwork package that is not present in the latest version of Magento 2.

**Comment by dpdplugin** [comment]
We have an update almost ready but we need a little bit more time.
Magento has changed the packaging system quite a bit, its not a simple class replacement.

**Comment by dpdplugin** [comment]
We are now in our testing phase, I expect a new version this week

**Comment by dpdplugin** [comment]
Fixed in 1.3.0
